### PR TITLE
Rewrite `immutable.Set/Map.+`

### PR DESCRIPTION
**Pull request by MasseGuillaume**
The `+` operation no longer has an overload accepting multiple values

motivation:
https://github.com/scala/collection-strawman/wiki/FAQ#what-are-the-breaking-changes

2nd row, breaking change

### CODE REVIEW

File: scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala [hunk 1]
```diff
@@ -156,6 +163,28 @@ case class Scalacollectioncompat_newcollections(index: SemanticdbIndex)
...
+  def replaceSetMapPlus2(ctx: RuleCtx): Patch = {
+    def rewritePlus(ap: Term.ApplyInfix, lhs: Term, op: Term.Name, rhs1: Term, rhs2: Term): Patch = {
+      ctx.replaceTree(
+        ap,
+        Term.ApplyInfix(
+          Term.ApplyInfix(lhs, op, Nil, List(rhs1)),
+          op,
+          Nil,
+          List(rhs2)
+        ).toString
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala]: What does `toString` do? That looks suspect :smile:

Comment (MasseGuillaume) [reply scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala]: It's fine, it will drop any comments. It's the scope of https://github.com/scalacenter/scala-syntax

Comment (julienrf) [reply scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala]: I was wondering whether `toString` was the recommended way to produce scala source code from a tree or if there was a method with a more meaningful name.

**Comment by MasseGuillaume** [comment]
@julienrf I don't see any corner case. `.syntax` takes care of everything. I added a test + fix for `(set + (2, 3)).map(x => x)` it was removing the `()` from the lhs. Nice catch!

**Comment by julienrf** [comment]
Thanks!
